core/src/main/java/org/mule/routing/filters/RegExFilter.java

```diff
@@ -117,21 +144,52 @@ public void setFlags(int flags)
         this.pattern = (this.pattern != null ? Pattern.compile(pattern.pattern(), flags) : null);
     }
 
+    public void setValue(String value)
+    {
+        this.value = new AttributeEvaluator(value);
+    }
+
     @Override
     public boolean equals(Object obj)
     {
-        if (this == obj) return true;
-        if (obj == null || getClass() != obj.getClass()) return false;
+        if (this == obj)
+        {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass())
+        {
+            return false;
+        }
 
         final RegExFilter other = (RegExFilter) obj;
         boolean patternsAreEqual = ClassUtils.equal(pattern.pattern(), other.pattern.pattern());
         boolean flagsAreEqual = (flags == other.flags);
-        return (patternsAreEqual && flagsAreEqual);
+        boolean equalValues = attributeWrapperRawValueEqualityValue(value, other.value);
+        return (patternsAreEqual && flagsAreEqual && equalValues);
+    }
+
+    @Override
+    public void setMuleContext(MuleContext context)
+    {
+        this.muleContext = context;
     }
 
     @Override
     public int hashCode()
     {
-        return hash(new Object[]{this.getClass(), pattern});
+        return hash(new Object[] {this.getClass(), pattern, value.getRawValue()});
+    }
+
+    private boolean attributeWrapperRawValueEqualityValue(AttributeEvaluator value1, AttributeEvaluator value2)
+    {
+        if (value1.getRawValue() == null && value2.getRawValue() == null)
```

Comment:
Isn't this the same as if (value1.getRawValue() == null) return value2.getRawValue() == null?